lib/turbolinks/redirection.rb

```diff
@@ -20,6 +20,26 @@ def redirect_to(url = {}, options = {})
       end
     end
 
+    def render(*args)
+      options = args.extract_options!
+      return super unless options[:turbolinks]
+
+      html = render_to_string args[0]
+      script = <<-SCRIPT
+        (function(){
+          Turbolinks.clearCache();
+          var parser = new DOMParser();
+          var doc = parser.parseFromString("#{ActionController::Base.helpers.j(html)}", "text/html");
+          document.documentElement.replaceChild(doc.body, document.body);
```

Comment:
Ideally this should be `Turbolinks.replace` or some other public method and not a hand-crafted replacement